### DISCUSSION TITLE
Delete multiple reports at once

### DISCUSSION THREAD

**Discussion by matanost-netop** [datapane/datapane#192] [Help] [answered]
Hi,
Is there a way to delete multiple reports at once, without individually going in to each of them and erasing it?
Something like common interfaces when one chooses multiple items from the gallery and press delete once to delete all.

Maybe, as an alternative, is there a way to delete reports prior to a certain date using CLI / python?
 
Thanks,
Matan.

**Comment by Buroni** [comment, marked as answer]
Hi Matan, you can delete reports before a certain date as follows:

```py
for details in dp.Report.list(): # List name + project of all reports belonging to me
    report = dp.Report.get(details["name"])
    timestamp = dateutil.parser.parse(report.created_date).timestamp()
    if timestamp < expiry_date: # `expiry_date` is some unix timestamp
        report.delete()
```
